### PR TITLE
Update preview version

### PR DESCRIPTION
**Pull request by srnagar**
To release a new parent pom version and a new preview version of core, the following changes were made:

1. Update parent pom version from 1.0.0 to 1.1.0
2. Update core to version to 1.0.0.preview.2
3. Update all sub-modules to use new parent pom version

### CODE REVIEW

**Comment by conniey** [comment]
Event Hubs has dependencies to azure-core and azure-core-amqp, should we also update those versions? https://github.com/Azure/azure-sdk-for-java/blob/8f2c7fb74ed9900b1d2f3ad5317a2f9d653090f7/eventhubs/client/pom.xml#L41

**Comment by srnagar** [comment]
@conniey Yes, we will update all client libraries except (KV and Identity) to use core preview 2 before releasing. I will have a separate PR for that. See @mayurid's comment on Teams.

**Comment by weshaggard** [comment]
@srnagar did you update these versions manually or did you use the the versions plugin (https://www.mojohaus.org/versions-maven-plugin/set-mojo.html). I only ask because using the plugin helps eliminate missed version bumps.

File: core/azure-core/pom.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
   <parent>
     <groupId>com.azure</groupId>
     <artifactId>azure-core-parent</artifactId>
-    <version>1.0.0-preview.1</version>
+    <version>1.0.0-preview.2</version>
     <relativePath>../pom.xml</relativePath>
   </parent>
```
Comment (weshaggard) [inline core/azure-core/pom.xml]: The version property should be explicitly set in this pom.xml file as well. It is actually an issue blocking the release that it currently isn't there. 

FYI @scbedd

File: core/azure-core-test/pom.xml [hunk 2]
```diff
@@ -5,14 +5,14 @@
   <parent>
     <groupId>com.azure</groupId>
```
Comment (weshaggard) [inline core/azure-core-test/pom.xml]: When we release the azure-core package we should avoid releasing the azure-core-test and other packages. We should only release the ones are libraries depend on which I believe is just azure-core and azure-core-amqp.

File: core/azure-core-amqp/pom.xml [hunk 3]
```diff
@@ -39,7 +39,7 @@
     <url>https://github.com/Azure/azure-sdk-for-java</url>
     <connection>scm:git:https://github.com/Azure/azure-sdk-for-java.git</connection>
     <developerConnection>scm:git:https://github.com/Azure/azure-sdk-for-java.git</developerConnection>
-    <tag>v1.0.0-preview.1</tag>
+    <tag>v1.0.0-preview.2</tag>
```
Comment (weshaggard) [inline core/azure-core-amqp/pom.xml]: I don't expect this tag is adding anything meaningful it is just duplication that can only get out of date. We should remove it here and in the other pom file that sets the tag to the version.

Comment (srnagar) [reply core/azure-core-amqp/pom.xml]: Removed the tag

File: core/azure-core/pom.xml (hunk 1 above)
Comment (srnagar) [reply core/azure-core/pom.xml]: Added the version.

**Comment by srnagar** [comment]
Thanks for the tip! I used the plugin to update the versions.

**Pull request merged**